docs/build-javadoc.sh

```diff
@@ -2,8 +2,9 @@
 
 set -ex
 pushd ../mlflow/java/client/
-mvn javadoc:javadoc
+mvn clean javadoc:javadoc
 popd
+rm -rf build/html/java_api/
```

Comment:
Delete the doc folder containing Javadocs (if it already exists) to clear out old files that have since been deleted